Build fails on mac os 10.14 due to not including cmath

Building https://github.com/llvm/llvm-project/blob/5927492e8e68c1ee5a0d84cf6c402a900aac4a3c/clang-tools-extra/clang-tidy/modernize/UseStdNumbersCheck.cpp#L46 throws error
```
/workspace/srcdir/llvm-project/clang-tools-extra/clang-tidy/modernize/UseStdNumbersCheck.cpp:46:10: error: call to 'abs' is ambiguous
--
return std::abs(Node.getValueAsApproximateDouble() - Value) < DiffThreshold;
       ^~~~~~~~
/opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/stdlib.h:132:6: note: candidate function
int      abs(int) __pure2;
         ^
/opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/c++/v1/stdlib.h:111:44: note: candidate function
inline _LIBCPP_INLINE_VISIBILITY long      abs(     long __x) _NOEXCEPT {return  labs(__x);}
                                          ^
/opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/c++/v1/stdlib.h:113:44: note: candidate function
inline _LIBCPP_INLINE_VISIBILITY long long abs(long long __x) _NOEXCEPT {return llabs(__x);}
```

Based on the sdks here https://github.com/joseluisq/macosx-sdks, it seems mac os 11.0 was the first to add `std::abs(double)` to `cstdlib` (which is currently included in the file).
To fix for earlier mac os versions including `cmath` works as noted in https://cplusplus.github.io/LWG/lwg-defects.html#2294.

--- a/clang-tools-extra/clang-tidy/modernize/UseStdNumbersCheck.cpp
+++ b/clang-tools-extra/clang-tidy/modernize/UseStdNumbersCheck.cpp
@@ -29,13 +29,13 @@
 #include "llvm/Support/FormatVariadic.h"
 #include "llvm/Support/MathExtras.h"
 #include <array>
+#include <cmath>
 #include <cstdint>
 #include <cstdlib>
 #include <initializer_list>
 #include <string>
 #include <tuple>
 #include <utility>
-#include <cmath>
 
 namespace {
 using namespace clang::ast_matchers;

--- a/clang-tools-extra/clang-tidy/modernize/UseStdNumbersCheck.cpp
+++ b/clang-tools-extra/clang-tidy/modernize/UseStdNumbersCheck.cpp
@@ -35,6 +35,7 @@
 #include <string>
 #include <tuple>
 #include <utility>
+#include <cmath>
 
 namespace {
 using namespace clang::ast_matchers;

